ksqldb-metastore/src/test/java/io/confluent/ksql/metastore/MetaStoreImplTest.java

```diff
@@ -116,30 +113,22 @@ public void shouldDeepCopyTypesOnCopy() {
   }
 
   @Test
-  public void shouldDeepCopySourceReferentialIntegrityDataOnCopy() {
+  public void shouldDeepCopyLinkedAndReferentialSourcesOnCopy() {
```

Comment:
I'm worried that it's not actually a deep copy. if we call `metaStore.linkSource()`, for example, on the copy - it will also add it to the source (see my comment above about deep copying). Can you add a test that covers this?

Reply:
Added. There wasn't any error, though.